testing/regress/hpcc/regression/cleanup.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import os
+import re
+import requests
+import json
+import time
+import glob
+
+logger = logging.getLogger('cleanup') 
+
+# Custom logger formatter for serial numbering of log records
+class SerialNumberFormatter(logging.Formatter):
+    def __init__(self):
+        super().__init__()
+        self.serialNumber = 0
+
+    def format(self, record):
+        self.serialNumber += 1
+        record.msg = str(self.serialNumber) + ". " + record.msg
+        return super().format(record)
+
+# Builds and configures the cleanup logger
+def buildCleanupLogger(logDir, cleanupLogger):
+    cleanupLogger.setLevel(logging.INFO)
+    curTime = time.strftime('%y-%m-%d-%H-%M-%S')
+    logName = "cleanup" + "." + curTime + ".log"
+    logPath = os.path.join(logDir, logName)
+    cleanupHandler = logging.FileHandler(logPath)
+    cleanupHandler.setFormatter(SerialNumberFormatter())
+    cleanupLogger.addHandler(cleanupHandler)
+    return cleanupLogger
+
+# Loads configuration data from JSON file
+def loadConfig(filename):
+    with open(filename, 'r') as file:
+        return json.load(file)
+
+# Extracts workunit details based on the cleanup mode
+def getRegressLogs(mode, logDirPath):
+    regressLogFiles = glob.glob(os.path.join(logDirPath, '*.log'))   
+    for logFilePath in regressLogFiles:
+        logFileNamePattern = r'^(thor|roxie|hthor)\.\d{2}-\d{2}-\d{2}-\d{2}-\d{2}-\d{2}\.log(?!.*exclusion)'
+        match = re.search(logFileNamePattern, os.path.basename(logFilePath))
+        if match:
+            if mode == "all" or mode == "workunits":
+                getWorkunitDetails(logFilePath)
+            elif mode == "passed":
+                getPassedWorkunitDetails(logFilePath)
+
+# Extracts workunit details from log file
+def getWorkunitDetails(logFilePath):
+    with open(logFilePath, 'r') as file:
+        lineList = file.readlines()
+        for line in lineList:
+            index = line.find("URL")
+            if index != -1:
+                url = line[index + 3:-1]
+                startIndex = url.find('Wuid=')
```

Comment:
The Wuid is being extracted so that it can be included in the log message after HTTP request for deletion. We can pass the url to the deleteWorkunits() and extract the Wuid within that function itself. That way the number of parameters being passed to deleteWorkunits() will be reduced.